src/identify/index.js

```diff
@@ -258,17 +306,73 @@ class IdentifyService {
       return log.error('received invalid message', err)
     }
 
-    // Update peers data in PeerStore
     const id = connection.remotePeer
+
+    // Legacy
+    if (!message.signedPeerRecord) {
+      try {
+        this.peerStore.addressBook.set(id, message.listenAddrs.map((addr) => multiaddr(addr)))
+      } catch (err) {
+        return log.error('received invalid listen addrs', err)
+      }
+
+      // Update the protocols
+      this.peerStore.protoBook.set(id, message.protocols)
+
+      return
+    }
+
+    // Open envelope and verify if is authenticated
+    let envelope
+    try {
+      envelope = await Envelope.openAndCertify(message.signedPeerRecord, PeerRecord.DOMAIN)
+    } catch (err) {
+      log('received invalid envelope, discard it')
+      throw errCode(new Error(messages.ERR_INVALID_ENVELOPE), codes.ERR_INVALID_ENVELOPE)
+    }
+
+    // Decode peer record
+    let peerRecord
+    try {
+      peerRecord = await PeerRecord.createFromProtobuf(envelope.payload)
+    } catch (err) {
+      log('received invalid peer record, discard it')
+      throw errCode(new Error(messages.ERR_INVALID_PEER_RECORD), codes.ERR_INVALID_PEER_RECORD)
+    }
+
+    // Update peers data in PeerStore
     try {
-      this.peerStore.addressBook.set(id, message.listenAddrs.map((addr) => multiaddr(addr)))
+      // TODO: Store as certified record
+
+      this.peerStore.addressBook.set(id, peerRecord.multiaddrs.map((addr) => multiaddr(addr)))
     } catch (err) {
       return log.error('received invalid listen addrs', err)
     }
 
     // Update the protocols
     this.peerStore.protoBook.set(id, message.protocols)
   }
+
+  /**
+   * Get self signed peer record raw envelope.
+   * @return {Buffer}
+   */
+  async _getSelfPeerRecord () {
+    // TODO: Verify if updated
```

Comment:
Make it easier to search the codebase for this later?
```suggestion
    // TODO: support invalidation when dynamic multiaddrs are supported
```